instrumentation/servlet/servlet-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/QueryParameterKeyCache.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.servlet;
+
+import io.opentelemetry.api.common.AttributeKey;
+import java.util.List;
+import java.util.Locale;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+class QueryParameterKeyCache {
+
+  public static final String KEY_PREFIX = "servlet.request.parameter.";
+  private static final ConcurrentMap<String, AttributeKey<List<String>>> cache =
+      new ConcurrentHashMap<>();
+
+  private QueryParameterKeyCache() {}
+
+  static AttributeKey<List<String>> get(String headerName) {
```

Comment:
should be `parameterName` instead of `headerName`?

Reply:
Yeah thanks, that was the old name. Will change it. 👍🏻